src/list.tsx

```diff
@@ -374,7 +374,7 @@ const useControlState = ({
             setIsChecked(!isChecked);
         }
         if (onChange) {
-            onChange(!value);
+            onChange(!isChecked);
```

Comment:
I think this should be:
```ts
onChange(isControlledByParent ? !value : !isChecked)
```
Please, add some tests to check both cases (controlled by parent and uncontrolled)

Reply:
in this case some simple unit tests should be enough